lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -205,6 +205,203 @@ sub schedules {
 
 =over 4
 
+=item update()
+
+Updates a job group according to the given YAML template. Test suites are added or modified
+as needed to reflect the difference to what's specified in the template.
+The given YAML will be validated and results in an error if it doesn't confom to the schema.
+
+Returns a 400 code on error, or a 303 code and the job template id within a JSON block on success.
+
+=back
+
+=cut
+
+sub update {
+    my $self = shift;
+
+    # No objects (aka SafeYAML)
+    $YAML::XS::LoadBlessed = 0;
+    my $yaml = YAML::XS::Load($self->param('template'));
+
+    my $errors;
+    eval { $errors = $self->app->validate_yaml($yaml, $ENV{MOJO_LOG_LEVEL} // '' =~ /DEBUG/i); };
```

Comment:
Indeed, checking `%ENV` here is wrong.

Reply:
Is it really ok to ignore `$@` here?

Reply:
I'm not sure that `$@` is ever set because JSON::Validator dies if the schema is erroneous... I added the `eval` in case upstream allows exceptions to be caught in the future.
But pushing `$@` to the `$errors` would seem sensible in any case.

Reply:
If you use `eval`, do something with `$@`. Ignoring it results in hard to find bugs in the future.

Reply:
Will move the error handling inside the `validate_yaml` helper.